torch/_dynamo/variables/torch.py

```diff
@@ -923,14 +966,18 @@ def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
 
             graph_checkpoint, checkpoint = tx.output.graph, tx.copy_graphstate()
 
-            sub_args = args[3].unpack_var_sequence(tx)
-
             def speculate_branch(branch):
-                # NB: 0 is predicate
-                ix = 1 if branch else 2
-                return speculate_subgraph(
-                    args[ix], sub_args, graph_checkpoint, checkpoint
-                )
+                try:
+                    # NB: 0 is predicate
+                    ix = 1 if branch else 2
+                    return speculate_subgraph(
+                        args[ix], operands, graph_checkpoint, checkpoint
+                    )
+                except TypeError as e:
+                    raise UserError(
+                        UserErrorType.COND_OP_RESTRICTION,
+                        f"{UserErrorType.COND_OP_RESTRICTION.name}. {str(e)}",
```

Comment:
oh wait, you store the code in UserError. So why doesn't UserError print it directly as part of rendering 🤔

Reply:
Yep. Will address this